libnmstate/nm/lldp.py

```diff
@@ -0,0 +1,330 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from libnmstate.nm import connection
+from libnmstate.schema import LLDP
+
+from .common import NM
+
+
+NM_VLAN_ID_KEY = "vid"
+NM_VLAN_NAME_KEY = "name"
+NM_MACPHY_AUTONEG_KEY = "autoneg"
+NM_MACPHY_PMD_AUTONEG_KEY = "pmd-autoneg-cap"
+NM_MACPHY_MAU_TYPE_KEY = "operational-mau-type"
+NM_PPVLAN_ID_KEY = "ppvid"
+NM_PPVLAN_FLAGS_KEY = "flags"
+NM_MANAGEMENT_ADDR_KEY = "address"
+NM_MANAGEMENT_ADDR_TYPE_KEY = "address-subtype"
+NM_MANAGEMENT_ADDR_IFACE_NUMBER_KEY = "interface-number"
+NM_MANAGEMENT_ADDR_IFACE_NUMBER_TYPE_KEY = "interface-number-subtype"
+NM_MANAGEMENT_ADDR_TYPE_IPV4 = 1
+NM_INTERFACE_TYPE_IFINDEX = 2
+NM_INTERFACE_TYPE_SYSTEM_PORT = 3
+
+
+LLDP_CAP_NAMES = [
+    LLDP.Neighbors.SYSTEM_CAPABILITY_OTHER,
+    LLDP.Neighbors.SYSTEM_CAPABILITY_REPEATER,
+    LLDP.Neighbors.SYSTEM_CAPABILITY_MAC_BRIDGE,
+    LLDP.Neighbors.SYSTEM_CAPABILITY_WLAN_AC,
+    LLDP.Neighbors.SYSTEM_CAPABILITY_ROUTER,
+    LLDP.Neighbors.SYSTEM_CAPABILITY_TELEPHONE,
+    LLDP.Neighbors.SYSTEM_CAPABILITY_DOCSIS,
+    LLDP.Neighbors.SYSTEM_CAPABILITY_STATION,
+    LLDP.Neighbors.SYSTEM_CAPABILITY_CVLAN,
+    LLDP.Neighbors.SYSTEM_CAPABILITY_SVLAN,
+    LLDP.Neighbors.SYSTEM_CAPABILITY_TPMR,
+]
+
+
+LLDP_CHASSIS_TYPE_TO_NMSTATE = {
```

Comment:
You may put 0 as `reserved` as spec file said and use list as above.

Reply:
Done, thanks!